test/framework/kogitoapp.go

```diff
@@ -30,61 +31,125 @@ const (
+)
+
+// KogitoAppDeployment defines the elements for the deployment of a kogito application
+type KogitoAppDeployment struct {
+	AppName     string
+	ContextDir  string
+	Runtime     KogitoAppDeploymentRuntime
+	Native      bool
+	Persistence bool
+	Events      bool
+	Labels      map[string]string
 }
 
-// DeploySpringBootExample deploys a Spring boot example
-func DeploySpringBootExample(namespace, appName, contextDir string, persistence, events bool, labels map[string]string) error {
-	GetLogger(namespace).Infof("Deploy spring boot example %s with name %s and persistence %v", contextDir, appName, persistence)
-	return DeployExample(namespace, appName, contextDir, "springboot", false, persistence, events, labels)
+// DeployExample deploy a Kogito example
+func DeployExample(namespace string, installerType InstallerType, kogitoAppDeployment KogitoAppDeployment) error {
+	GetLogger(namespace).Infof("%s deploy %s example %s with name %s, native %v, persistence %v, events %v and labels %v", installerType, kogitoAppDeployment.Runtime, kogitoAppDeployment.ContextDir, kogitoAppDeployment.AppName, kogitoAppDeployment.Native, kogitoAppDeployment.Persistence, kogitoAppDeployment.Events, kogitoAppDeployment.Labels)
+	switch installerType {
+	case CLIInstallerType:
+		return cliDeployExample(namespace, kogitoAppDeployment)
+	case CRInstallerType:
+		return crDeployExample(namespace, kogitoAppDeployment)
+	default:
+		return fmt.Errorf("DeployExample: Unknown installer type %s", installerType.Name)
+	}
 }
 
 // DeployExample deploys an example
-func DeployExample(namespace, appName, contextDir, runtime string, native, persistence, events bool, labels map[string]string) error {
-	kogitoApp := getKogitoAppStub(namespace, appName, labels)
-	if runtime == "quarkus" {
+func crDeployExample(namespace string, kogitoAppDeployment KogitoAppDeployment) error {
+	kogitoApp := getKogitoAppStub(namespace, kogitoAppDeployment.AppName, kogitoAppDeployment.Labels)
+	if kogitoAppDeployment.Runtime == QuarkusKogitoAppDeploymentRuntime {
 		kogitoApp.Spec.Runtime = v1alpha1.QuarkusRuntimeType
-	} else if runtime == "springboot" {
+	} else if kogitoAppDeployment.Runtime == SpringBootKogitoAppDeploymentRuntime {
 		kogitoApp.Spec.Runtime = v1alpha1.SpringbootRuntimeType
 	}
 
 	gitProjectURI := GetConfigExamplesRepositoryURI()
-	kogitoApp.Spec.Build.Native = native
+	kogitoApp.Spec.Build.Native = kogitoAppDeployment.Native
 	kogitoApp.Spec.Build.GitSource.URI = &gitProjectURI
-	kogitoApp.Spec.Build.GitSource.ContextDir = contextDir
+	kogitoApp.Spec.Build.GitSource.ContextDir = kogitoAppDeployment.ContextDir
 	kogitoApp.Spec.Build.GitSource.Reference = GetConfigExamplesRepositoryRef()
 
 	// Add namespace for service discovery
 	// Can be removed once https://issues.redhat.com/browse/KOGITO-675 is done
 	appendNewEnvToKogitoApp(kogitoApp, "NAMESPACE", namespace)
 
-	profiles := ""
-	if persistence {
-		profiles = profiles + "persistence,"
+	profiles := []string{}
```

Comment:
Please consider using `var profile []string` instead of allocating an empty array in memory.